client/src/lib/supabase/middleware.ts

```diff
@@ -1,5 +1,51 @@
 import { createServerClient } from "@supabase/ssr";
 import { NextResponse, type NextRequest } from "next/server";
+import { userHasExistingProfile } from "@/lib/actions/profile_setup"
+
+//debug utility function 
+const DEBUG = false;
+export function debug(msg: string){
+  if (DEBUG){
+    console.log("DEBUG: ", msg);
+  }
+}
+
+debug("***** BEGIN middleware.ts ******");
+debug("middleware.ts is always invoked!!!");
+
+//allowed public paths
+const publicPaths = ["/","/login","/explore","/signup","/about"];
+
+//whitelists auth'd user paths
+const validUserPaths = ["/message", "/profile", "/thread", "/track", "/profile_setup"];
```

Comment:
Is it possible to target the files based on the directory instead? It seems hard to maintain all routes if we're targeting each route

Reply:
(main) and (onboarding) are not referenceable by path since the parathesis exclude them from pathnames